src/test/java/uk/gov/digital/ho/pttg/api/AuditHistoryResourceTest.java

```diff
@@ -200,6 +200,60 @@ public void getAllCorrelationIds_returnedCorrelationIds_logCount() {
         assertRequestDurationLogged(loggingEvent);
     }
 
+    @Test
+    public void getAllCorrelationIds_withToDate_callService() {
+        List<AuditEventType> someEventTypes = Arrays.asList(INCOME_PROVING_FINANCIAL_STATUS_REQUEST, INCOME_PROVING_FINANCIAL_STATUS_RESPONSE);
+        LocalDate someDate = LocalDate.now();
+
+        historyResource.getAllCorrelationIds(someEventTypes, someDate);
+
+        then(mockHistoryService).should().getAllCorrelationIds(someEventTypes, someDate);
+    }
+
+    @Test
+    public void getAllCorrelationIds_withToDate_returnCorrelationIdsFromService() {
+        List<String> expectedCorrelationIds = Arrays.asList("some correlation id", "some other correlation id");
+        given(mockHistoryService.getAllCorrelationIds(anyList(), any(LocalDate.class)))
+                .willReturn(expectedCorrelationIds);
+
+        List<AuditEventType> anyEventTypes = Collections.singletonList(INCOME_PROVING_FINANCIAL_STATUS_REQUEST);
+        LocalDate anyDate = LocalDate.now();
+        List<String> actualCorrelationIds = historyResource.getAllCorrelationIds(anyEventTypes, anyDate);
+
+        assertThat(actualCorrelationIds).isEqualTo(expectedCorrelationIds);
+    }
+
+    @Test
+    public void getAllCorrelationIds_withToDate_logEntryParameters() {
+        List<AuditEventType> eventTypes = Arrays.asList(INCOME_PROVING_FINANCIAL_STATUS_REQUEST, INCOME_PROVING_FINANCIAL_STATUS_RESPONSE);
+        LocalDate toDate = LocalDate.parse("2019-07-30");
+
+        historyResource.getAllCorrelationIds(eventTypes, toDate);
+
+        then(mockAppender).should(atLeastOnce()).doAppend(logCaptor.capture());
+
+        String expectedLogMessage = String.format("Requested all correlation ids for events [%s, %s] up to %s",
```

Comment:
I'm still not keen on testing for exact messages. Better to test for contains message parameters only?